File: python/ABSTRACT_FACTORY_DECORATOR/ABSTRACT_FACTORY_DECORATOR/abstract_factory/ReservacionSencillaFactory.py
from ABSTRACT_FACTORY_DECORATOR.reservacion.ReservacionSencilla import ReservacionSencilla
from .ReservacionFactory import ReservacionFactory

class ReservacionSencillaFactory(ReservacionFactory):
    
    reservacionSencilla : ReservacionSencilla
    
    def crearReservacion(self, datos):
        self.crearReservacionSencilla(datos)
    
    def crearReservacionSencilla(self, datos):
        self.reservacionSencilla = ReservacionSencilla(datos)
    

File: php/ABSTRACT_FACTORY_DECORATOR/abstractFactory/ReservacionSencillaFactory.php
<?php

require_once './abstractFactory/ReservacionFactory.php';
require_once './reservacion/ReservacionSencilla.php';

class ReservacionSencillaFactory implements ReservacionFactory{
    
    public $reservacionSencilla;
    
    public function crearReservacion($datos) {
        $this->crearReservacionSencilla($datos);
    }
    
    public function crearReservacionSencilla($datos) {
        $this->reservacionSencilla= new ReservacionSencilla($datos);
    }

}
